instance/tasks.py

```diff
@@ -60,3 +63,51 @@ def spawn_appserver(instance_ref_id, mark_active_on_success=False, num_attempts=
                 # finish and replace the second as the active server. We are not really worried about that for now.
                 instance.set_appserver_active(appserver_id)
             break
+
+
+@db_task()
+def shut_down_instances():
+    """
+    Shut down instances whose PRs got merged (more than) one week ago.
+
+    Terminate all app servers belonging to these instances, and disable monitoring.
+    """
+    for instance in OpenEdXInstance.objects.filter(watchedpullrequest__isnull=False):
+        pr = github.get_pr_info_by_number(
+            instance.watchedpullrequest.target_fork_name,
+            instance.watchedpullrequest.github_pr_number
+        )
+        if pr['state'] == 'closed':
+            closed_at = github.parse_date(pr['closed_at'])
+            now = datetime.now()
+            if sufficient_time_passed(closed_at, now, 7):
+                instance.disable_monitoring()
+                for appserver in instance.appserver_set.all():
+                    appserver.terminate_vm()
+
+
+@db_task()
+def terminate_obsolete_appservers():
+    """
+    Terminate app servers that were created (more than) two days before the currently-active app server
+    of each individual instance.
+
+    Do nothing for instances that don't have an active app server.
+    """
+    for instance in OpenEdXInstance.objects.all():
+        active_appserver = instance.active_appserver
+        if active_appserver:
+            for appserver in instance.appserver_set.all():
+                if sufficient_time_passed(appserver.created, active_appserver.created, 2):
+                    appserver.terminate_vm()
```

Comment:
@itsjeyd Could lines 98-102 be moved to a `instance.terminate_obsolete_servers(days=2)` method? (And maybe rename this task into `terminate_all_instances_obsolete_appservers()`?